packages/wallets/src/api/client.ts

```diff
@@ -0,0 +1,302 @@
+import { validateAPIKey, environmentToCrossmintBaseURL, APIKeyUsageOrigin } from "@crossmint/common-sdk-base";
+import { type Client, createClient, createConfig } from "@hey-api/client-fetch";
+
+import type {
+    ClientOptions,
+    CreateWalletDto,
+    WalletV1Alpha2ResponseDto,
+    CreateTransactionDto,
+    SubmitApprovalDto,
+    WalletsV1ControllerGetTransaction4Response,
+    WalletsV1Alpha2TransactionResponseDto,
+    CreateSignatureRequestDto,
+    WalletsV1Alpha2SignatureResponseDto,
+    WalletsV1Alpha2TransactionsResponseDto,
+    Nftevm,
+    Nftsol,
+    WalletBalanceResponseDto,
+} from "./gen/types.gen";
+import {
+    walletsV1ControllerCreateWallet4,
+    walletsV1ControllerGetWalletByLocator4,
+    walletsV1ControllerCreateTransaction4,
+    walletsV1ControllerSubmitApprovals4,
+    walletsV1ControllerGetTransaction4,
+    walletsV1ControllerCreateSignatureRequest4,
+    walletsV1ControllerSubmitSignatureApprovals4,
+    walletsV1ControllerGetSignature4,
+    walletsV1ControllerGetTransactionsWithoutChain4,
+    fetchContentFromWallet,
+    balanceControllerGetBalanceForLocator2,
+} from "./gen/sdk.gen";
+
+type CreateWalletParams = CreateWalletDto;
+type CreateWalletResponse = WalletV1Alpha2ResponseDto;
+type GetWalletResponse = WalletV1Alpha2ResponseDto;
+
+type CreateTransactionParams = CreateTransactionDto;
+type CreateTransactionResponse = WalletsV1Alpha2TransactionResponseDto;
+type ApproveTransactionParams = SubmitApprovalDto;
+type ApproveTransactionResponse = WalletsV1ControllerGetTransaction4Response;
+type GetTransactionResponse = WalletsV1Alpha2TransactionResponseDto;
+
+type CreateSignatureParams = CreateSignatureRequestDto;
+type CreateSignatureResponse = WalletsV1Alpha2SignatureResponseDto;
+type ApproveSignatureParams = SubmitApprovalDto;
+type ApproveSignatureResponse = WalletsV1Alpha2SignatureResponseDto;
+type GetSignatureResponse = WalletsV1Alpha2SignatureResponseDto;
+
+type GetTransactionsResponse = WalletsV1Alpha2TransactionsResponseDto;
+type GetNftsResponse = Nftevm | Nftsol;
+type GetBalanceResponse = WalletBalanceResponseDto;
+
+class ApiClient {
+    private readonly client: Client;
+
+    constructor(
+        private readonly apiKey: string,
+        private readonly jwt?: string
```

Comment:
the jwt can change over time (e.g. when the user refreshes the token). im not sure this ap client supports jwt mutations?

Reply:
Didn't know it can change, thanks for context!

No it doesn't support that. But note that the SDK doesn't support that as well (I know there was a discussion on supporting that, but I haven't seen a specific API proposed in the PRD). So it doesn't limit us, at least for now.

Reply:
I'm not sure i follow!

The JWT can mutate without an API - it updates whenever we refresh the token, which is an internal operation of the auth SDK.

Reply:
The initial SDK spec didn't account for that, and this implementation doesn't either.

However, we're thinking of adopting most/all functionality from `CrossmintAuth`, which will include JWT refresh.